packages/appinio_swiper/lib/appinio_swiper.dart

```diff
@@ -1,679 +1,743 @@
+        duration: duration,
+      ),
+      curve: curve,
+      begin: position.offset,
+      end: target,
+    );
+    await _startActivity(newActivity);
+  }
+
+  @override
+  void initState() {
+    super.initState();
 
-    _animationController.addStatusListener((status) {
-      //when status of controller changes
-      if (status == AnimationStatus.completed) {
-        setState(() {
-          if (_swipeType == 1) {
-            _swiperMemo[currentIndex] = _horizontal
-                ? (_swipedDirectionHorizontal == 1
-                    ? AppinioSwiperDirection.right
-                    : AppinioSwiperDirection.left)
-                : (_swipedDirectionVertical == 1
-                    ? AppinioSwiperDirection.top
-                    : AppinioSwiperDirection.bottom);
-            _swipedDirectionHorizontal = 0;
-            _swipedDirectionVertical = 0;
-            _horizontal = false;
-            if (widget.loop) {
-              if (currentIndex < widget.cardsCount - 1) {
-                currentIndex++;
-              } else {
-                currentIndex = 0;
-              }
-            } else {
-              currentIndex++;
-            }
-            widget.onSwipe?.call(currentIndex, detectedDirection);
-            if (currentIndex == widget.cardsCount) {
-              widget.onEnd?.call();
-            }
-          } else if (_swipeType == 2) {
-            _isUnswiping = false;
-          }
-          _animationController.reset();
-          _left = 0;
-          _top = 0;
-          _total = 0;
-          _angle = 0;
-          _scale = 0.9;
-          _difference = widget.cardsSpacing;
-          _swipeType = 0;
-        });
-      }
+    WidgetsBinding.instance.addPostFrameCallback((timeStamp) {
+      // Attach the controller after the frame because `_attach` uses `position`
+      // which isn't valid until after `initState` has finished.
+      widget.controller?._attach(this);
     });
   }
 
   @override
   void dispose() {
     super.dispose();
-    _animationController.dispose();
+    widget.controller?._detach();
+    _swipeActivity?.animation.dispose();
   }
 
   @override
-  Widget build(BuildContext context) {
-    return LayoutBuilder(
-      builder: (BuildContext context, BoxConstraints constraints) {
-        return Container(
-          padding: widget.padding,
-          child: LayoutBuilder(
-            builder: (BuildContext context, BoxConstraints constraints) {
-              return Stack(
-                  clipBehavior: Clip.none,
-                  fit: StackFit.expand,
-                  children: [
-                    if (widget.loop || currentIndex < widget.cardsCount - 1)
-                      ..._backgroundCards(constraints),
-                    if (currentIndex < widget.cardsCount)
-                      _foregroundItem(constraints)
-                  ]);
-            },
-          ),
-        );
-      },
-    );
+  void didUpdateWidget(covariant AppinioSwiper oldWidget) {
```

Comment:
Since we save a lot of the widget configurations in our position object we need to update the position object whenever the relevant configuration changes.